cluster-autoscaler/cloudprovider/alicloud/alicloud_manager.go

```diff
@@ -107,7 +107,7 @@ func (m *AliCloudManager) GetAsgSize(asgConfig *Asg) (int64, error) {
 	if err != nil {
 		return -1, fmt.Errorf("failed to describe ASG %s,Because of %s", asgConfig.id, err.Error())
 	}
-	return int64(sg.ActiveCapacity + sg.PendingCapacity), nil
+	return int64(sg.DesiredCapacity), nil
```

Comment:
I think it should be compatible with both enabled 'Expected number of instances' and disabeld situations.

Maybe this is an option:
```suggestion
	return int64(max(sg.DesiredCapacity, sg.ActiveCapacity + sg.PendingCapacity)), nil
```

Reply:
This is incorrect. Firstly, there are syntax errors. Additionally, configuring desiredSize is necessary. It should prompt rather than be compatible. Moreover, compatibility with downsizing could pose risks, so I don’t recommend this change.